microprofile-metrics/README.adoc

```diff
@@ -0,0 +1,987 @@
+`http://localhost:9990/metrics` endpoint using your browser or
+`curl http://localhost:9990/metrics`.
+
+NOTE: Don't forget to have {productName} server started.
+
+You will get back a text file similar to this:
+
+[source]
+----
+# HELP base_cpu_processCpuLoad Displays the "recent cpu usage" for the Java Virtual Machine process.
+# TYPE base_cpu_processCpuLoad gauge
+base_cpu_processCpuLoad 0.005753138075313807
+# HELP base_memory_committedNonHeap_bytes Displays the amount of memory that is committed for the Java virtual machine to use.
+# TYPE base_memory_committedNonHeap_bytes gauge
+base_memory_committedNonHeap_bytes 8.9915392E7
+----
+
+This is a Prometheus format which is directly consumable by
+the https://prometheus.io/[Prometheus monitoring system]. This is a default
+format. If you prefer for your testing to use rather the JSON format you need to
+specify the HTTP header `Accepted: application/json` with for
+instance `curl -H "Accept: application/json" http://localhost:9990/metrics`.
+
+== Different scopes of metrics
+
+The MicroProfile Metrics specification defines three different scopes of metrics:
+
+* Base scope (`/metrics/base`)- the metrics that all MicroProfile vendors have to
+provide
+* Vendor scope (`/metrics/vendor`) - vendor specific metrics
+* Application scope (`/metrics/application`) - application specific metrics
+
+Base scope metrics cover the metrics that are generally useful in any environment
+like `memory.usedHeap` or `thread.count`. For the full list of required metrics
+please access the
+`http://localhost:9990/metrics/base` endpoint using your browser or
+`curl http://localhost:9990/metrics/base`.
+
+Vendor metrics represent custom metrics that are provided by the implementation of
+the MicroProfile Metrics specification which are in our case the custom metrics
+provided by the {productName} server. To see the full list please access
+the `http://localhost:9990/metrics/vendor` endpoint using your browser or
+`curl http://localhost:9990/metrics/vendor`. You can note that there are custom
+metrics exposed by the individual {productName} subsystems which are prefixed by
+ `wildfly_` or `jboss_`.
+
+The application scope covers the metrics defined by your application. We will
+demonstrate how to define application metrics in the following sections.
+
+== OPTIONS requests for metrics metadata
+
+So far we've accessed the metrics only with the HTTP GET requests which will always
+return the individual values of the requested metrics. The specification also provides
+an OPTIONS method requests for users to request the metrics metadata:
+
+Perform a request
+`curl -X OPTIONS -H "Accept: application/json" http://localhost:9990/metrics` and you
+can see that the returned JSON now contains metadata about individual metrics like
+unit, type, description, display name and tags. We will see later on how you can
+change these values for your custom applications scoped metrics.
+
+== Application scoped metrics
```

Comment:
"application scoped" makes me think it's somehow connected to CDI scopes, but it's not, I'd remove the "scoped". Or maybe "application-specific" would do